WordPress/Classes/ViewRelated/Site Creation/Wizard/SiteCreator.swift

```diff
@@ -36,23 +35,40 @@ final class SiteCreator {
     ///
     /// - Returns: an Encodable object
     ///
-    func build() throws -> SiteCreationRequest {
-        guard let domainSuggestion = address else {
-            throw SiteCreationRequestAssemblyError.invalidDomain
-        }
-        let siteName = domainSuggestion.isWordPress ? domainSuggestion.subdomain : domainSuggestion.domainName
-        let siteDesign = design?.slug ?? "default"
+    func build() -> SiteCreationRequest {
 
         let request = SiteCreationRequest(
             segmentIdentifier: segment?.identifier,
-            siteDesign: siteDesign,
+            siteDesign: design?.slug ?? Strings.defaultDesignSlug,
             verticalIdentifier: vertical?.slug,
-            title: information?.title ?? NSLocalizedString("Site Title", comment: "Site info. Title"),
+            title: information?.title ?? Strings.defaultSiteTitle,
             tagline: information?.tagLine ?? "",
-            siteURLString: siteName,
-            isPublic: true
+            siteURLString: siteURLString,
+            isPublic: true,
+            siteCreationFlow: address == nil ? Strings.siteCreationFlowForNoAddress : nil,
+            findAvailableURL: address == nil
         )
-
         return request
     }
+
+    var hasSiteTitle: Bool {
+        information?.title != nil
+    }
+
+    /// Returns the domain suggestion if there's one,
+    /// - otherwise a site name if there's one,
+    /// - otherwise an empty string.
+    private var siteURLString: String {
+
+        guard let domainSuggestion = address else {
+            return information?.title ?? ""
+        }
+        return domainSuggestion.isWordPress ? domainSuggestion.subdomain : domainSuggestion.domainName
+    }
+
+    private enum Strings {
+        static let defaultDesignSlug = "default"
+        static let defaultSiteTitle = NSLocalizedString("Site Title", comment: "Site info. Title")
```

Comment:
I think the default `Site Title` will be set by the backend even if we don't pass it. I'm not suggesting a change :)

Reply:
Nice! Yeah, I'd vote to keep this since it's what we set from the client before Site Name (the current flow).

Reply:
oh yeah thanks for catching this! That title has no reason to be there I guess